cmd/vulcan-repository-sctrl/custom-rules/repository-with-security-control.yaml

```diff
@@ -0,0 +1,112 @@
+rules:
+  - id: repository-with-lava-binary
+    message: 'Action using lava binary'
+    severity: INFO
+    languages:
+      - yaml
+    paths:
+      include:
+        - ".github/**/*.yaml"
+        - ".github/**/*.yml"
+    patterns:
+      - pattern-regex: 'lava scan'
+      - pattern-regex: 'lava run'
+  - id: repository-with-lava-action
+    message: 'Action running Lava'
+    severity: INFO
+    languages:
+      - yaml
+    paths:
+      include:
+        - ".github/**/*.yaml"
+        - ".github/**/*.yml"
+    patterns:
+      - patterns:
+          - pattern: |
+              ...
+              uses: "$USES"
+          - metavariable-pattern:
+              metavariable: $USES
+              language: generic
+              pattern-either:
+                - pattern: lava-internal-action
+                - pattern: lava-action
+  - id: repository-with-sonarqube-action
+    message: 'Action running Sonarqube'
+    severity: INFO
+    languages:
+      - yaml
+    paths:
+      include:
+        - ".github/**/*.yaml"
+        - ".github/**/*.yml"
+    patterns:
+      - pattern: |
+          ...
+          uses: "$USES"
+      - metavariable-pattern:
+          metavariable: $USES
+          language: generic
+          pattern: code-quality-action
+  - id: repository-with-trivy-action
+    message: 'Action running Trivy'
+    severity: INFO
+    languages:
+      - yaml
+    paths:
+      include:
+        - ".github/**/*.yaml"
+        - ".github/**/*.yml"
+    patterns:
+      - pattern: |
+          ...
+          uses: "$USES"
+      - metavariable-pattern:
+          metavariable: $USES
+          language: generic
+          pattern: trivy-action
+  - id: repository-with-snyk-action
+    message: 'Action running Snyk'
+    severity: INFO
+    languages:
+      - yaml
+    paths:
+      include:
+        - ".github/**/*.yaml"
+        - ".github/**/*.yml"
+    patterns:
+      - pattern: |
+          ...
+          uses: "$USES"
+      - metavariable-pattern:
+          metavariable: $USES
+          language: generic
+          pattern: snyk/actions
+  - id: repository-with-govulncheck
+    message: 'Action running Govulncheck'
+    severity: INFO
+    languages:
+      - yaml
+    paths:
+      include:
+        - ".github/**/*.yaml"
+        - ".github/**/*.yml"
+    patterns:
+      - pattern: |
+          ...
+          uses: "$USES"
+      - metavariable-pattern:
+          metavariable: $USES
+          language: generic
+          pattern: govulncheck-action
+  - id: repository-with-govulncheck-binary
+    message: 'Action using govulncheck binary'
+    severity: INFO
+    languages:
+      - yaml
+    paths:
+      include:
+        - ".github/**/*.yaml"
+        - ".github/**/*.yml"
+    patterns:
+      - pattern-regex: 'govulncheck \./\.\.\.'
```

Comment:
I feel this rule is a bit "weak".  But let's keep it like this and see how well it performs.